fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/ProcedureDef.java

```diff
@@ -44,17 +40,23 @@ public void apply(IDatabaseAdapter target) {
         }
     }
 
-    /* (non-Javadoc)
-     * @see com.ibm.fhir.database.utils.model.IDatabaseObject#drop(com.ibm.fhir.database.utils.api.IDatabaseAdapter)
-     */
+    @Override
+    public void apply(Integer priorVersion, IDatabaseAdapter target) {
+        // Serialize the execution of the procedure, to try and avoid the
+        // horrible deadlocks we keep getting
+        synchronized(target) {
+            // TODO migration logic here?
+            target.createProcedure(getSchemaName(), getObjectName(), supplier);
+        }
+    }
+
     @Override
     public void drop(IDatabaseAdapter target) {
         target.dropProcedure(getSchemaName(), getObjectName());
     }
-    
-       @Override
-       protected void grantGroupPrivileges(IDatabaseAdapter target, Set<Privilege> group, String toUser) {
-            target.grantProcedurePrivileges(getSchemaName(), getObjectName(), group, toUser);
-        }
 
+    @Override
+    protected void grantGroupPrivileges(IDatabaseAdapter target, Set<Privilege> group, String toUser) {
+        target.grantProcedurePrivileges(getSchemaName(), getObjectName(), group, toUser);
```

Comment:
we should leave a note here to make it intentional that grants should be carefully evaluated to see if there is an update required in the Migration step, such as a change to the operational user.

Reply:
not sure why this code is showing as changed...all I did is fix the indentation.

do you think having that comment here makes sense, or is there a spot that makes more sense?
maybe at the class level on ProcedureDef?